src/Catty/DataModel/VariablesContainer/VariablesContainer.h

```diff
@@ -76,6 +77,8 @@
 - (SpriteObject*)spriteObjectForObjectVariable:(UserVariable*)userVariable;
 
 - (BOOL)isProjectVariableOrList:(UserVariable*)userVariable;
```

Comment:
Please remove this now deprecated method.

Reply:
Hi @m-herold if this function is removed then what should be the condition for the `if statement` in the function `- (BOOL)isVarOrListBeingUsed:(UserVariable*)variable`

should it be `[self.object.project.variables isProjectVariable:variable]` ?

Reply:
or should the condition be `[self.object.project.variables isProjectVariable:variable || [self.object.project.variables isProjectList:variable` but the function `isProjectList:` uses a `UserList` instead of `UserVariable`.

Should I use `UserVariable` in  the function `isProjectList:` too (for now)?
Please correct me if I am doing something wrong.

Reply:
Are you talking about the isVarOrListBeingUsed function of the `FormulaElement`?

Reply:
yes

Reply:
Are you sure? I can not see that this function is used there. Having a closer look at the code reveals that `isProjectVariableOrList` is used in a function with the same name but in the `FormulaEditorViewController`. I assume you mean that one, right?

Reply:
Yes it is used in `FormulaEditorViewController` not in `FormulaElement`

Reply:
Very good question. Please consider the `UserList` as a not-ready feature which is currently disabled by a feature toggle. Instead of making one very large branch which is hard to maintain and almost impossible to review when handing in one very large pull request, we are making several small pull requests for this major refactoring. This should speed up the time-to-merge dramatically and ease code reviews, etc.

This means that for now, please do not use `UserList` in the `isProjectList` function yet, since the actual implementation is still a `UserVariable`. I will make the comments also directly to the relevant lines of code.

Reply:
Back to your question: Yes, please make use of `isProjectVariable:variable || isProjectList:variable` or similar.